src/iSparrow/sparrow_watcher.py

```diff
@@ -0,0 +1,527 @@
+from iSparrow import SparrowRecording
+from iSparrow import SpeciesPredictorBase
+import iSparrow.utils as utils
+
+from pathlib import Path
+import pandas as pd
+from watchdog.events import FileSystemEventHandler
+from watchdog.observers import Observer
+from time import sleep
+from datetime import datetime
+from copy import deepcopy
+import yaml
+import multiprocessing
+
+
+class AnalysisEventHandler(FileSystemEventHandler):
+    """
+    AnalysisEventHandler Custom event handler that calls a function whenever
+    the a new file with a matching extension is created in the watched directory and that waits for a
+    multiprocessing.Event object to be true.
+
+    Base:
+        FileSystemEventHandler: watchdog.FilesystemEventHandler
+
+    Methods:
+    --------
+    on_created(event): Run the callback whenever a newly created file with a matching pattern is found.
+    """
+
+    def __init__(
+        self,
+        watcher,
+    ):
+        """
+        __init__ Create a new AnalysisEventHandler object.
+
+        Args:
+            callback (callable): Callback functionw hen
+            pattern (str, optional): file apttern. Defaults to ".wav".
+            watcher: (SparrowWatcher): Watcher this Handler is used with
+        """
+        self.pattern = watcher.pattern
+        self.recording = watcher.set_up_recording()
```

Comment:
it uses the setup capabilities of the watcher, but all the modules for the models etc is loaded  only here, which allows them to live in the watcher process and not the main process. (the latter approach created problems on macos).